src/bin.tsx

```diff
@@ -16,36 +19,61 @@ program
   .storeOptionsAsProperties(false)
   .passCommandToAction(false)
   .description('Deletes old GitLab pipelines')
-  .arguments('<gitlab-url>')
-  .arguments('<project-id>')
-  .arguments('<access-token>')
+  .arguments('[gitlab-url]')
+  .arguments('[project-id]')
+  .arguments('[access-token]')
   .option('-d --days <days>', 'older than days', '30')
   .option('--trace', 'show stack traces for errors when possible', false)
-  .action((gitlabUrl: string, projectId: string, accessToken: string, options: Record<string, unknown>) => {
-    const parsedProjectId = parseInt(projectId, 10);
-    const days = typeof options.days === 'string' ? parseInt(options.days, 10) : 30;
-    const startDate = new Date();
-    const listPipelinesFunction = listPipelines({ getRequest, gitlabUrl, projectId: parsedProjectId, accessToken });
-    const deletePipelineFunction = deletePipeline({
-      deleteRequest,
-      gitlabUrl,
-      projectId: parsedProjectId,
-      accessToken,
-    });
-    render(
-      <Main
-        gitlabUrl={gitlabUrl}
-        projectId={parsedProjectId}
-        accessToken={accessToken}
-        days={days}
-        startDate={startDate}
-        exit={exit}
-        listPipelines={listPipelinesFunction}
-        filterPipelinesByDate={filterPipelinesByDate}
-        deletePipeline={deletePipelineFunction}
-        showStackTraces={options.trace === true}
-      />,
-    );
-  });
+  .action(
+    async (
```

Comment:
I think for `async` actions it is required to use `program.parseAsync` instead of `program.parse`.

Reply:
Good point 👍 I didn't knew about that function.